Dockerfile

```diff
@@ -1,6 +1,17 @@
 # CONTAINER FOR BUILDING BINARY
 FROM golang:1.21 AS build
 
+# TODO: REMOVE SSH BEFORE MERGING, ONCE BEETHOVEN IS PUBLIC
+
+# SSH KEY
+ARG SSH_PRIVATE_KEY
```

Comment:
There is another option to do it via git token

Dockerfile:
```
ENV GOPRIVATE="github.com/0xPolygon/beethoven"
ADD scripts/git-configure.sh .gittoken ./
RUN mkdir -p -m 0600 ~/.ssh && \
  ssh-keyscan github.com >> ~/.ssh/known_hosts && \
  sh ./git-configure.sh
```

scripts/git-configure.sh
```
#!/bin/sh

# This script is needed to support using packages from private repos.
# This script applies the given github token if exists, otherwise tells to use ssh instead of https for downloading repos.

export GITHUB_TOKEN=$(cat .gittoken)

if [[ -z "${GITHUB_TOKEN}" ]]; then
    git config --global url.ssh://git@github.com/.insteadOf https://github.com/
else
    git config --global url."https://$GITHUB_TOKEN:x-oauth-basic@github.com/".insteadOf "https://github.com/"
fi
```

.gittoken (ignorred)
```
<git-token>
```

Also just FYI